### PR TITLE
Implement /marker command

### LINKED ISSUE #1109: Add /marker command (from twitch chat)
Context: https://help.twitch.tv/s/article/creating-highlights-and-stream-markers?language=en_US#Add

### Steps to reproduce

- Go to a channel you own
- Start streaming
- Send `/marker` command 

### Observed

System message that says `Sorry, "/user" is not available through this client.`

2019-06-22-220115_611x141_scrot

### Expected

Stream marker is successfully added.

### PR DESCRIPTION
**Pull request by zneix**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Requires chatterino/login#10

Command was tested in webchat and I managed to replicate its behaviour in 100%.

Closes #1109

### CODE REVIEW

**Comment by Mm2PL** [comment]
Please add a case for anon users :-)

**Comment by zneix** [comment]
> Please add a case for anon users :-)

Good idea, added it in 7573d59

**Review by Mm2PL — APPROVED** [review]

File: src/providers/twitch/api/README.md [hunk 1]
```diff
@@ -107,6 +107,14 @@ Requires `user:edit:follows` scope
      * `widgets/dialogs/UserInfoPopup.cpp` to unfollow a user by unticking follow checkbox in usercard
      * `controllers/commands/CommandController.cpp` in /unfollow command
 
+### Create Stream Marker
+URL: https://dev.twitch.tv/docs/api/reference/#create-stream-marker
```
Comment (pajlada) [inline src/providers/twitch/api/README.md:111]:
```suggestion
URL: https://dev.twitch.tv/docs/api/reference/#create-stream-marker  
```
add two spaces here so it shows up nicely on a new line

File: src/providers/twitch/api/README.md [hunk 2]
```diff
@@ -107,6 +107,14 @@ Requires `user:edit:follows` scope
...
+Requires `user:edit:broadcast` scope
+
+ * We implement this in `providers/twitch/api/Helix.cpp unfollowUser`
```
Comment (pajlada) [inline src/providers/twitch/api/README.md:114]:
```suggestion
 * We implement this in `providers/twitch/api/Helix.cpp unfollowUser`  
```

Adds two new lines here so it shows up nicely on a new line

File: src/controllers/commands/CommandController.cpp [hunk 3]
```diff
@@ -452,6 +453,48 @@ void CommandController::initialize(Settings &, Paths &paths)
...
+    this->registerCommand("/marker", [](const QStringList &words,
+                                        auto channel) {
+        // Avoid Helix calls without Client ID and/or OAuth Token
+        if (getApp()->accounts->twitch.getCurrent()->isAnon())
+        {
+            channel->addMessage(makeSystemMessage(
+                "You need to be logged in to create stream markers!"));
+        }
+
+        auto *twitchChannel = dynamic_cast<TwitchChannel *>(channel.get());
```
Comment (pajlada) [inline src/controllers/commands/CommandController.cpp:466]: Please check the return value of `dynamic_cast` to ensure this doesn't crash. Even if non-`TwitchChannel`s don't currently use `CommandController`, it's still a sane thing to do

File: src/controllers/commands/CommandController.cpp [hunk 4]
```diff
@@ -452,6 +453,48 @@ void CommandController::initialize(Settings &, Paths &paths)
 
             return "";
         });
+
+    this->registerCommand("/marker", [](const QStringList &words,
+                                        auto channel) {
+        // Avoid Helix calls without Client ID and/or OAuth Token
+        if (getApp()->accounts->twitch.getCurrent()->isAnon())
+        {
+            channel->addMessage(makeSystemMessage(
```
Comment (pajlada) [inline src/controllers/commands/CommandController.cpp:462]: You're missing a return here to actually stop the request from happening if the user is not logged in

File: src/controllers/commands/CommandController.cpp [hunk 5]
```diff
@@ -452,6 +453,48 @@ void CommandController::initialize(Settings &, Paths &paths)
...
+        }
+
+        auto *twitchChannel = dynamic_cast<TwitchChannel *>(channel.get());
+
+        // Exact same message as in webchat
+        if (!twitchChannel->isLive())
+        {
+            channel->addMessage(makeSystemMessage(
+                "You can only add stream markers during live streams. Try "
+                "again when the channel is live streaming."));
```
Comment (pajlada) [inline src/controllers/commands/CommandController.cpp:473]: Missing return here - we shouldn't even try creating a stream marker if the channel is live.

**Review by pajlada — CHANGES_REQUESTED** [review]

File: src/controllers/commands/CommandController.cpp (hunk 3 above)
Comment (zneix) [reply src/controllers/commands/CommandController.cpp:467]: I am a little confused because every command is using this solution everywhere.

**Comment by zneix** [comment]
We just tested it with @Mm2PL on livestream and triple-checked scopes of a token that we're using for creating stream marker and scope `user:edit:broadcast` doesn't seem to be required for marker creation. A possible Twitch bug?

**Review by pajlada — APPROVED** [review]

**Comment by pajlada** [comment]
Thank you for your contribution! :heart:

**Pull request merged**